src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,256 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private static final User userNull = null;
+
+    @BeforeAll
+    public static void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    public void registerAndGetNewUser_Ok() {
+        User userFirst = createUser(22, "crimeabridgedestroyed", "Himars");
+        User userSecond = createUser(70, "freedom", "Bayraktar");
+        registrationService.register(userFirst);
+        User expectedUser1 = storageDao.get(Storage.people.get(0).getLogin());
+        assertEquals(userFirst, expectedUser1);
+        registrationService.register(userSecond);
+        User expectedUser2 = storageDao.get(Storage.people.get(1).getLogin());
+        assertEquals(userSecond, expectedUser2);
+    }
+
+    @Test
+    void registerAndGetDifferentUsers_Ok() {
+        User userFirst = createUser(22, "crimeabridgedestroyed", "Himars");
+        User userSecond = createUser(70, "freedom", "Bayraktar");
+        registrationService.register(userFirst);
+        registrationService.register(userSecond);
+        User expectedUser1 = storageDao.get(Storage.people.get(0).getLogin());
+        User expectedUser2 = storageDao.get(Storage.people.get(1).getLogin());
+        assertNotEquals(expectedUser1, expectedUser2);
+    }
+
+    @Test
+    void userNull_GetExceptionMessage_Ok() {
+        try {
+            registrationService.register(userNull);
+        } catch (NullPointerException e) {
+            String actualMessage = e.getMessage();
+            String expectedMessage = "User should not be null";
+            assertEquals(actualMessage, expectedMessage);
+        }
+    }
+
+    @Test
+    void userOneDouble_ExceptionMessage_Ok() {
+        User userFirst = createUser(22, "crimeabridgedestroyed", "Himars");
+        User userSecond = createUser(70, "freedom", "Bayraktar");
+        registrationService.register(userFirst);
+        registrationService.register(userSecond);
+        User userDoubleFirst = createUser(22, "crimeabridgedestroyed", "Himars");
```

Comment:
repeated data for valid user can be constants, like that
    public static final String DEFAULT_LOGIN = "login";
    public static final String DEFAULT_PASSWORD = "password";
    public static final int DEFAULT_AGE = 30;